internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -355,6 +357,70 @@ func resourceComputeClusterRead(d *pluginsdk.ResourceData, meta interface{}) err
 	return tags.FlattenAndSet(d, computeResource.Model.Tags)
 }
 
+func resourceComputeClusterUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MachineLearning.MachineLearningComputes
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := machinelearningcomputes.ParseComputeID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	compute, err := client.ComputeGet(ctx, *id)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	computeModel := compute.Model
+	if computeModel == nil {
+		return fmt.Errorf("retrieving %s: `model` was nil", *id)
+	}
+
+	identity, err := expandIdentity(d.Get("identity").([]interface{}))
+	if err != nil {
+		return fmt.Errorf("expanding `identity`: %+v", err)
+	}
+
+	vmPriority := machinelearningcomputes.VMPriority(d.Get("vm_priority").(string))
+	computeClusterAmlComputeProperties := machinelearningcomputes.AmlComputeProperties{
+		VMSize:                 utils.String(d.Get("vm_size").(string)),
+		VMPriority:             &vmPriority,
+		ScaleSettings:          expandScaleSettings(d.Get("scale_settings").([]interface{})),
+		UserAccountCredentials: expandUserAccountCredentials(d.Get("ssh").([]interface{})),
+		EnableNodePublicIP:     pointer.To(d.Get("node_public_ip_enabled").(bool)),
+	}
+
+	computeClusterAmlComputeProperties.RemoteLoginPortPublicAccess = pointer.To(machinelearningcomputes.RemoteLoginPortPublicAccessDisabled)
+	if d.Get("ssh_public_access_enabled").(bool) {
+		computeClusterAmlComputeProperties.RemoteLoginPortPublicAccess = pointer.To(machinelearningcomputes.RemoteLoginPortPublicAccessEnabled)
+	}
+
+	if subnetId, ok := d.GetOk("subnet_resource_id"); ok && subnetId.(string) != "" {
+		computeClusterAmlComputeProperties.Subnet = &machinelearningcomputes.ResourceId{Id: subnetId.(string)}
+	}
+
+	computeClusterProperties := machinelearningcomputes.AmlCompute{
+		Properties:       &computeClusterAmlComputeProperties,
+		ComputeLocation:  utils.String(d.Get("location").(string)),
+		Description:      utils.String(d.Get("description").(string)),
+		DisableLocalAuth: utils.Bool(!d.Get("local_auth_enabled").(bool)),
+	}
+
+	computeClusterParameters := machinelearningcomputes.ComputeResource{
+		Properties: computeClusterProperties,
+		Identity:   identity,
+		Location:   computeModel.Location,
+		Tags:       tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if err := client.ComputeCreateOrUpdateThenPoll(ctx, *id, computeClusterParameters); err != nil {
+		return fmt.Errorf("updating %s: %+v", id, err)
+	}
```

Comment:
@xuzhang3 [my original comment](https://github.com/hashicorp/terraform-provider-azurerm/pull/26404#discussion_r1648653670) hasn't been resolved here. 

Like we do in many other updates, we get the existing resource, and patch in the changes to the existing resource's model and use that as the payload for the CreateOrUpdate call. This is explained in our [guide for adding new resources in the Contributor Docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md).

This whole block should be simplified to

```suggestion
	existing, err := client.ComputeGet(ctx, *id)
	if err != nil {
		return fmt.Errorf("retrieving %s: %+v", *id, err)
	}

	payload := existing.Model
	if payload == nil {
		return fmt.Errorf("retrieving %s: `model` was nil", *id)
	}

	identity, err := expandIdentity(d.Get("identity").([]interface{}))
	if err != nil {
		return fmt.Errorf("expanding `identity`: %+v", err)
	}

	payload.Identity = identity

	if err := client.ComputeCreateOrUpdateThenPoll(ctx, *id, *payload); err != nil {
		return fmt.Errorf("updating %s: %+v", id, err)
	}
```

Reply:
> @xuzhang3 [my original comment](https://github.com/hashicorp/terraform-provider-azurerm/pull/26404#discussion_r1648653670) hasn't been resolved here.
> 
> Like we do in many other updates, we get the existing resource, and patch in the changes to the existing resource's model and use that as the payload for the CreateOrUpdate call. This is explained in our [guide for adding new resources in the Contributor Docs](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md).
> 
> This whole block should be simplified to

update as requested and all the tests passed.